### PR TITLE
Delete of an unknown package fails with a wrong error

### LINKED ISSUE #13: Delete of an unknown package fails with a wrong error
Keep getting a {"code":"InternalError","message":"Object.keys called on non-object"} error whenever I try to delete a package

```
 curl -X DELETE http://localhost:8080/package/mypackage/0.0.1
```

### PR DESCRIPTION
**Pull request by paolodm**
Fixes #13

### CODE REVIEW

**Comment by bajtos** [comment]
Hi @paolodm, thank you for submitting a pull request with a bugfix.

Please reset the git history in your forked repository, as the pull request contains a copy of commits that were already merged into master (e.g. 4b4e789). Also don't merge master branch to you feature branch, use `git rebase` instead as I mentioned elsewhere. This way the pull request won't contain merge commits like 65042c4.

Having done that, please add a unit-test for your change.

**Comment by paolodm** [comment]
Added unit test and rebased from mbrevoort/master.

File: test/data.js [hunk 1]
```diff
@@ -20,5 +20,17 @@ describe('Data', function() {
         done();
       });
     });
+
+    describe('an unpublished package', function() {
+      it('reports that package does not exist', function() {
```
Comment (bajtos) [inline test/data.js]: 1. The goal of  BDD-style test names is to get a nice sentence specifying program's behaviour. This test yields
   
   ```
   Data deletePackage() an unpublished package reports that package does not exist.
   ```
   
   I am proposing to get rid of the extra describe and use one of these names:
   
   ```
     it('reports error when package does not exist', function() { // etc.
     it('reports that package does not exist for an unpublished package', function() {// etc.
   ```
2. The test does not check that "package does not exist" was reported, it only verifies that an exception was not thrown.

**Comment by paolodm** [comment]
Removed the extra describe block as prescribed.

**Comment by bajtos** [comment]
> Removed the extra describe block as prescribed.

Cool. Please address the second issue too:

> The test does not check that "package does not exist" was reported, it only verifies that an exception was not thrown.

IMO the best way of testing this change is to move test's expectation to deletePackage callback and check what error was reported.

**Comment by paolodm** [comment]
Bajtos, 

> IMO the best way of testing this change is to move test's expectation to deletePackage callback and check what error was reported.

This has been implemented and pushed to PR.

**Comment by bajtos** [comment]
Landed as f544fd6, released in v0.1.8. Thank you for your contribution.

**Pull request closed without merging**